### PR TITLE
Add VectorWork work type

### LINKED ISSUE #211: Migrate VectorWork
- [ ] Model / metadata
- [ ] ChangeSet
- [ ] Controller
- [ ] Decorator

### PR DESCRIPTION
**Pull request by hackmastera**
fixes #211, refs #223

### CODE REVIEW

**Comment by hackmastera** [comment]
@eliotjordan do we need to add `VectorWork` to this list? https://github.com/pulibrary/figgy/blob/0fe25d1811cc6d0df43295a564753c70a8aa0786/app/models/ability.rb#L145

**Comment by eliotjordan** [comment]
@hackmastera Yes we do. Good catch.

**Comment by hackmastera** [comment]
@eliotjordan cool I will add it thanks

File: app/controllers/concerns/browse_everythingable.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+module BrowseEverythingable
```
Comment (tpendragon) [inline app/controllers/concerns/browse_everythingable.rb]: Should we just add this to `Valhalla::ResourceController`?

File: app/decorators/concerns/member_of_collections_display.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+module MemberOfCollectionsDisplay
```
Comment (tpendragon) [inline app/decorators/concerns/member_of_collections_display.rb]: This makes me pretty uncomfortable. I'm not willing to stop the PR over it, but I wonder if we need a different pattern.

Comment (tpendragon) [reply app/decorators/concerns/member_of_collections_display.rb]: (I don't know what that pattern is - maybe some sort of global property rendering registration?)

Comment (tpendragon) [reply app/decorators/concerns/member_of_collections_display.rb]: In fact, maybe it's better to just add `member_of_collections` to the default decorator and have the individual decorators add it to `display_attributes` if they need it?

File: app/decorators/ephemera_template_decorator.rb [hunk 3]
```diff
@@ -2,12 +2,6 @@
 class EphemeraTemplateDecorator < Valkyrie::ResourceDecorator
   self.display_attributes = []
 
-  delegate :query_service, to: :metadata_adapter
-
-  def metadata_adapter
```
Comment (tpendragon) [inline app/decorators/ephemera_template_decorator.rb]: All of this moving `metadata_adapter` to the default decorator is really great, btw.

File: app/controllers/concerns/browse_everythingable.rb (hunk 1 above)
Comment (eliotjordan) [reply app/controllers/concerns/browse_everythingable.rb]: Some controllers don't make use of BrowseEverything, `EphemeraFieldsController` for instance, but do include `Valhalla::ResourceController`. If that isn't an issue, then this seems reasonable. Alternately, we could put this into `BaseResourceController`.

File: app/decorators/concerns/member_of_collections_display.rb (hunk 2 above)
Comment (eliotjordan) [reply app/decorators/concerns/member_of_collections_display.rb]: I think the default decorator is a good spot for `member_of_collections`.

File: app/models/scanned_map.rb [hunk 4]
```diff
@@ -15,11 +15,16 @@ def self.can_have_manifests?
     true
   end
 
+  # How is this used?
```
Comment (tpendragon) [inline app/models/scanned_map.rb]: This is a good question - I'm not sure it is anymore. It used to be the label on display in a lot of places for Hyrax, but I think that's a different method in the decorator now.

**Comment by tpendragon** [comment]
The PR looks great to me. Couple questions about the use of modules, but those can just be tickets we create.

File: app/controllers/concerns/browse_everythingable.rb (hunk 1 above)
Comment (tpendragon) [reply app/controllers/concerns/browse_everythingable.rb]: I'm not sure it's a problem. I wonder if there's some sort of trigger or organization we should be thinking about which is along the lines of "has filesets" and "doesn't have filesets"

**Comment by hackmastera** [comment]
Re: browse_everythingable and member_of_collections_display: in both cases I argued against including the behavior in a base class since not all inheriting classes would use the behaviors. I'm not sure why that bugs me! But I will accept the collective "it's cool" and get over it :)

**Comment by hackmastera** [comment]
I got rid of those two modules. @tpendragon what do you think of the BaseResourceController in general? would you prefer to see all this as part of Valhalla::ResourceController?
 I wasn't sure how to make that decision. https://github.com/pulibrary/figgy/blob/441e8e39613258ad14620c694fb10c1653d40e9b/app/controllers/base_resource_controller.rb

**Comment by tpendragon** [comment]
I'm not sure, really. It's true not everything needs `browse_everything`. Maybe keep the BaseResourceController and see how it goes over time?

**Comment by hackmastera** [comment]
@tpendragon that sounds good. In which case I think this PR is ready to go, unless there's something I lost track of.

**Pull request merged**